python/do.py

```diff
@@ -0,0 +1,199 @@
+        memgraph, if_query if condition else else_query, parameters=params
+    )
+
+    return [
+        mgp.Record(
+            value={
+                field_name: _gqlalchemy_type_to_mgp(ctx.graph, field_value)
+                for field_name, field_value in result.items()
+            }
+        )
+        for result in results
+    ]
+
+
+def _get_edge_by_id(
+    graph: mgp.Graph, edge_id: mgp.EdgeId, from_id: mgp.VertexId
+) -> mgp.Nullable[mgp.Edge]:
+    return next(
+        (
+            edge
+            for edge in graph.get_vertex_by_id(from_id).out_edges
+            if edge.id == edge_id
+        ),
+        None,
+    )
+
+
+def _gqlalchemy_type_to_mgp(graph: mgp.Graph, variable: Any) -> mgp.Any:
+    if isinstance(
+        variable,
+        (
+            bool,
+            int,
+            float,
+            str,
+            list,
+            dict,
+            datetime.date,
+            datetime.time,
+            datetime.datetime,
+            datetime.timedelta,
+        ),
+    ):
+        return variable
+
+    elif isinstance(variable, gqlalchemy.models.Node):
+        return graph.get_vertex_by_id(int(variable._id))
+
+    elif isinstance(variable, gqlalchemy.models.Relationship):
+        return _get_edge_by_id(
+            graph, edge_id=int(variable._id), from_id=int(variable._start_node_id)
+        )
+
+    elif isinstance(variable, gqlalchemy.models.Path):
+        start_id = int(variable._nodes[0]._id)
+        path = mgp.Path(graph.get_vertex_by_id(start_id))
+
+        for relationship, node in zip(variable._relationships, variable._nodes[1:]):
+            next_edge_id = int(relationship._id)
+            next_node_id = int(node._id)
+
+            edge_to_add = _get_edge_by_id(graph, edge_id=next_edge_id, from_id=start_id)
+            if edge_to_add is None:
+                # This should happen if and only if the query that returned the path treated graph edges as undirected.
+                # For example, such a query might return an (a)-[b]->(c) path from a graph that only contains (c)->[b].
+                # In that case, flipping the path direction to retrieve the linking edge is valid.
+                edge_to_add = _get_edge_by_id(
+                    graph, edge_id=next_edge_id, from_id=next_node_id
+                )
+
+            path.expand(edge_to_add)
+
+            start_id = next_node_id
+
+        return path
+
+
+# Conditional queries are run via gqlalchemy. The module supports parameterized
+# queries, which are only supported from gqlalchemy 1.4.0. Since that version’s
+# dgl requirement causes arm64 checks for MAGE to fail, this function serves to
+# support running parameterized queries with older gqlalchemy versions.
+#
+# Relevant code:
+# https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/vendors/database_client.py#L54-L59
+# https://github.com/memgraph/gqlalchemy/blob/main/gqlalchemy/connection.py#L87-L100
+def _execute_and_fetch_parametrized(
```

Comment:
This is nice that you implemented this since you needed for the debugging but I think we cannot use this since this gqlalchemy code. Did you check if python3.8 supports DGL on arm? Maybe that's not even an issue, it was just an idea that popped into my mind. Did you also check how is DGL built? If I am not wrong we added some part to be built from source. Because I really think is conceptually wrong to merge gqlalchemy functionalities with mage modules

Reply:
Also, please mention this on the daily so we can discuss it. I think this gives us enough evidence that we need a unified Python version on all Memgraph related things -> gqla, mage and Memgraph, otherwise there will be whole mess of dependencies in the future.

Reply:
The docs don’t mention anything about ARM; I’ve only found that people have run dgl on ARM machines. Where do I find how we build dgl?

Reply:
I agree with you about it being conceptually wrong to merge gqlalchemy features with mage, but this workaround is supposed to be there only until https://github.com/memgraph/gqlalchemy/issues/227 is solved.

Reply:
You can check yml files in workflow directory and Dockerfile

Reply:
Do you know how this issue should be resolved? How can we make optional dependencies

Reply:
The GQLAlchemy team is already aware of the optional dependency issue :heavy_check_mark:

Reply:
Since we concluded at the daily that this isn’t a blocker, can this be resolved?
When GQLAlchemy with optional dependencies is shipped, I can make a quick patch to remove this workaround.

Reply:
in my opinion yes, when gqla is fixed, patch here should suffice